JSONConstant.java

```diff
@@ -65,7 +65,7 @@ public String toString() {
   public boolean equals(Object other) {
     return ( ( (other instanceof JSONConstant) 
                && (this.value == ((JSONConstant) other).value) )
-             || (this.value == other) );
+             || (this == other) );
```

Comment:
`equals()` isn't supposed to check if the references are equal, it's supposed to check if the values are equal.